src/bsoncxx/CMakeLists.txt

```diff
@@ -64,8 +64,17 @@ add_library(bsoncxx_static STATIC
 
 set_target_properties(bsoncxx_static PROPERTIES
     OUTPUT_NAME bsoncxx
+    COMPILE_DEFINITIONS "BSONCXX_STATIC"
```

Comment:
Use target_compile_definitions(bsoncxx_static PUBLIC BSONCXX_STATIC), it is clearer